Fix (make robust) test_timer_context_cancelled under Python 3.7

## Long story short

<!-- Please describe your problem and why the fix is important. -->
With emerging of 3.7, we need to adjust test suite to pass under this env

## Expected behaviour

<!-- What is the behaviour you expect? -->
test passes

## Actual behaviour

One of coroutine cancellation tests fails:
<!-- What's actually happening? -->
```python-traceback
$ make cov-ci-no-ext
...
――――――――――――――――――――――――― test_timer_context_cancelled ―――――――――――――――――――――――――
    def test_timer_context_cancelled():
        with mock.patch('aiohttp.helpers.asyncio') as m_asyncio:
            m_asyncio.TimeoutError = asyncio.TimeoutError
            loop = mock.Mock()
            ctx = helpers.TimerContext(loop)
            ctx.timeout()
    
            with pytest.raises(asyncio.TimeoutError):
                with ctx:
                    pass
    
>           assert m_asyncio.Task.current_task.return_value.cancel.called
E           AssertionError: assert False
E            +  where False = <MagicMock name='asyncio.Task.current_task().cancel' id='23073207868104'>.called
E            +    where <MagicMock name='asyncio.Task.current_task().cancel' id='23073207868104'> = <MagicMock name='asyncio.Task.current_task()' id='23073207876128'>.cancel
E            +      where <MagicMock name='asyncio.Task.current_task()' id='23073207876128'> = <MagicMock name='asyncio.Task.current_task' id='23073207863616'>.return_value
E            +        where <MagicMock name='asyncio.Task.current_task' id='23073207863616'> = <MagicMock name='asyncio.Task' id='23073207799992'>.current_task
E            +          where <MagicMock name='asyncio.Task' id='23073207799992'> = <MagicMock name='asyncio' id='23073210485560'>.Task
tests/test_helpers.py:430: AssertionError
```

## Steps to reproduce

<!-- Please describe steps to reproduce the issue.
     If you have a script that does that please include it here within
     markdown code markup -->
See https://travis-ci.com/aio-libs/aiohttp/jobs/130964628#L793-L814

## Your environment

<!-- Describe the environment you have that lead to your issue.
     This includes aiohttp version, OS, proxy server and other bits that
     are related to your case. 
     
     IMPORTANT: aiohttp is both server framework and client library.
     For getting rid of confusing please put 'server', 'client' or 'both'
     word here.
     -->
Travis CI

--- a/.travis.yml
+++ b/.travis.yml
@@ -135,9 +135,8 @@ os: linux
 jobs:
   fast_finish: true
   allow_failures:
-  - python:
-    - 3.7-dev
-    - nightly
+  - python: 3.7-dev
+  - python: nightly
 
   include:
   - <<: *_doc_base

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,6 @@
 version: "= 0"
 
 sudo: false
-dist: trusty
 
 language: python
 
@@ -136,7 +135,9 @@ os: linux
 jobs:
   fast_finish: true
   allow_failures:
-  - python: nightly
+  - python:
+    - 3.7-dev
+    - nightly
 
   include:
   - <<: *_doc_base

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,7 @@
 version: "= 0"
 
 sudo: false
+dist: trusty
 
 language: python
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,7 @@ language: python
 python:
 - 3.5
 - &mainstream_python 3.6
-- 3.7-dev
+- 3.7
 - nightly
 - &pypy3 pypy3.5
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,7 +8,7 @@ language: python
 python:
 - 3.5
 - &mainstream_python 3.6
-- 3.7
+- 3.7-dev
 - nightly
 - &pypy3 pypy3.5
 

--- a/tests/test_connector.py
+++ b/tests/test_connector.py
@@ -493,9 +493,13 @@ async def test_tcp_connector_certificate_error(loop):
 
     assert isinstance(ctx.value, ssl.CertificateError)
     assert isinstance(ctx.value.certificate_error, ssl.CertificateError)
+    if PY_37:
+        err_cls = 'SSLCertVerificationError'
+    else:
+        err_cls = 'CertificateError'
     assert isinstance(ctx.value, aiohttp.ClientSSLError)
-    assert str(ctx.value).startswith('Cannot connect to host 127.0.0.1:443 '
-                                     'ssl:True')
+    assert str(ctx.value) == ('Cannot connect to host 127.0.0.1:443 ssl:True '
+                              '[{}: ()]'.format(err_cls))
 
 
 async def test_tcp_connector_multiple_hosts_errors(loop):

--- a/tests/test_connector.py
+++ b/tests/test_connector.py
@@ -493,13 +493,7 @@ async def test_tcp_connector_certificate_error(loop):
 
     assert isinstance(ctx.value, ssl.CertificateError)
     assert isinstance(ctx.value.certificate_error, ssl.CertificateError)
-    if PY_37:
-        err_cls = 'SSLCertVerificationError'
-    else:
-        err_cls = 'CertificateError'
     assert isinstance(ctx.value, aiohttp.ClientSSLError)
-    assert str(ctx.value) == ('Cannot connect to host 127.0.0.1:443 ssl:True '
-                              '[{}: ()]'.format(err_cls))
 
 
 async def test_tcp_connector_multiple_hosts_errors(loop):

--- a/tests/test_connector.py
+++ b/tests/test_connector.py
@@ -493,13 +493,9 @@ async def test_tcp_connector_certificate_error(loop):
 
     assert isinstance(ctx.value, ssl.CertificateError)
     assert isinstance(ctx.value.certificate_error, ssl.CertificateError)
-    if PY_37:
-        err_cls = 'SSLCertVerificationError'
-    else:
-        err_cls = 'CertificateError'
     assert isinstance(ctx.value, aiohttp.ClientSSLError)
-    assert str(ctx.value) == ('Cannot connect to host 127.0.0.1:443 ssl:True '
-                              '[{}: ()]'.format(err_cls))
+    assert str(ctx.value).startswith('Cannot connect to host 127.0.0.1:443 '
+                                     'ssl:True')
 
 
 async def test_tcp_connector_multiple_hosts_errors(loop):

--- a/tests/test_connector.py
+++ b/tests/test_connector.py
@@ -19,6 +19,7 @@ from aiohttp import client, web
 from aiohttp.client import ClientRequest, ClientTimeout
 from aiohttp.client_reqrep import ConnectionKey
 from aiohttp.connector import Connection, _DNSCacheTable
+from aiohttp.helpers import PY_37
 from aiohttp.test_utils import make_mocked_coro, unused_port
 from aiohttp.tracing import Trace
 
@@ -492,9 +493,13 @@ async def test_tcp_connector_certificate_error(loop):
 
     assert isinstance(ctx.value, ssl.CertificateError)
     assert isinstance(ctx.value.certificate_error, ssl.CertificateError)
+    if PY_37:
+        err_cls = 'SSLCertVerificationError'
+    else:
+        err_cls = 'CertificateError'
     assert isinstance(ctx.value, aiohttp.ClientSSLError)
     assert str(ctx.value) == ('Cannot connect to host 127.0.0.1:443 ssl:True '
-                              '[CertificateError: ()]')
+                              '[{}: ()]'.format(err_cls))
 
 
 async def test_tcp_connector_multiple_hosts_errors(loop):
@@ -2021,12 +2026,19 @@ async def test_tcp_connector_raise_connector_ssl_error(aiohttp_server):
     session = aiohttp.ClientSession(connector=conn)
     url = srv.make_url('/')
 
-    with pytest.raises(aiohttp.ClientConnectorSSLError) as ctx:
-        print(url)
+    if PY_37:
+        err = aiohttp.ClientConnectorCertificateError
+    else:
+        err = aiohttp.ClientConnectorSSLError
+    with pytest.raises(err) as ctx:
         await session.get(url)
 
-    assert isinstance(ctx.value.os_error, ssl.SSLError)
-    assert isinstance(ctx.value, aiohttp.ClientSSLError)
+    if PY_37:
+        assert isinstance(ctx.value, aiohttp.ClientConnectorCertificateError)
+        assert isinstance(ctx.value.certificate_error, ssl.SSLError)
+    else:
+        assert isinstance(ctx.value, aiohttp.ClientSSLError)
+        assert isinstance(ctx.value.os_error, ssl.SSLError)
 
     await session.close()
 

--- a/tests/test_helpers.py
+++ b/tests/test_helpers.py
@@ -427,7 +427,10 @@ def test_timer_context_cancelled():
             with ctx:
                 pass
 
-        assert m_asyncio.Task.current_task.return_value.cancel.called
+        if helpers.PY_37:
+            assert m_asyncio.current_task.return_value.cancel.called
+        else:
+            assert m_asyncio.Task.current_task.return_value.cancel.called
 
 
 def test_timer_context_no_task(loop):

